### ISSUE TITLE
[feature] Attempt package download from other remotes

### ISSUE THREAD

**Issue by Ignition** [conan-io/conan#6028]
Currently when you have multiple remotes where each have a common recipe. Once the recipe is found in one remote that is the only remote checked when downloading the required package. 

It would be preferable to attempt downloading from all remotes which have a copy of that recipe.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @Ignition 

This is true when using packages without revisions, and it makes sense for that case. If you have multiple repos where you can have the same package reference, it is very possible that recipes for the same reference become out-of-sync, that is why the default behavior always tried to get the packages belonging to the server from where the recipe was downloaded. That behavior can be force to retrieve packages from other servers, if the remote is explicitly defined with ``-r=remote``

The behavior you are requesting is already implemented for package revisions. As the revision is either the recipe contents or the commit hash, in either case it is safe to assume that the recipes of different remote servers will be in sync, and thus, the active remotes are iterated in search of the binary to be installed. Please try to enable the package revisions and check.

Also, instead of having multiple remotes, it is possible (and typically a good practice) to use the virtual repositories in Artifactory to aggregate those other repos, and let developers use a single remote as the single source of packages.

**Comment by Ignition** [comment]
So looking at revisions. Our artifactory instance is new enough that it is supported. But we need out client to have the revisions enabled to change their behaviour for recipe export and package creation, such that the uploaded recipe and packages on the remote are revisioned. Is that correct?

Assuming I've understood, at the moment packages on the server are RREV 0. Conan clients currently are uploading and overriding that RREV 0 because they are not using `revisions_enabled=1`. Can `revisions_enabled=0` clients still freely override once revisions recipe and packages have been uploaded?

**Comment by memsharded** [comment]
> But we need out client to have the revisions enabled to change their behaviour for recipe export and package creation, such that the uploaded recipe and packages on the remote are revisioned. Is that correct?

Yes, you need to define ``revisions_enabled`` in *conan.conf* to activate them in the client. (https://docs.conan.io/en/latest/versioning/revisions.html#how-to-activate-the-revisions) (suggest to manage conan.conf with ``conan config install``)

Assuming I've understood, at the moment packages on the server are RREV 0. Conan clients currently are uploading and overriding that RREV 0 because they are not using revisions_enabled=1. Can revisions_enabled=0 clients still freely override once revisions recipe and packages have been uploaded?

Yes, the revision 0 is "just" another revision, it is not incompatible with the others. Revisions are time-ordered in the server the moment they are uploaded. If a client with revisions disabled builds something and uploads it, it will be with revision=0, will be uploaded, overwrite what was there, and the revision 0 will become the latest one, and the default to be installed by consumers of any type (both with revisions and without revisions). Note that revisions=0 will also always fetch the latest revisions, not necessarily the revision=0.

This was the way revisions could be introduced with backwards compatibility and a reasonable behavior.
